staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -147,3 +147,35 @@ func ComputeHistogramDelta(before, after model.Samples, label model.LabelName) {
 func makeKey(a, b model.LabelValue) string {
 	return string(a) + "___" + string(b)
 }
+
+// GetMetricValuesForLabel returns value of metric for a given dimension
+func GetMetricValuesForLabel(ms Metrics, metricName, label string) map[string]int64 {
+	result := make(map[string]int64)
+	for method, samples := range ms {
+		if method != metricName {
+			continue
+		}
+		for _, sample := range samples {
+			count := int64(sample.Value)
+			dimensionName := string(sample.Metric[model.LabelName(label)])
+			result[dimensionName] = count
+		}
+	}
+	return result
+}
+
+// ValidateMetrics verifies if every sample of metric has all expected labels
+func ValidateMetrics(metrics Metrics, metricName string, expectedLabels ...string) error {
+	samples, ok := metrics[metricName]
+	if !ok {
+		return fmt.Errorf("Metric %q was not found in metrics", metricName)
```

Comment:
nit: error strings should start with lower case

Reply:
Done